onnx/version_converter/adapters/axis_input_to_attribute.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) ONNX Project Contributors
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "onnx/version_converter/adapters/adapter.h"
+
+namespace ONNX_NAMESPACE {
+namespace version_conversion {
+class AxisInputToAttribute : public Adapter {
+ public:
+  explicit AxisInputToAttribute(
+      const std::string& op_name,
+      const OpSetID& initial,
+      const OpSetID& target,
+      size_t axis_index,
+      int64_t default_axis)
+      : Adapter(op_name, initial, target), axis_index(axis_index), default_axis(default_axis) {}
+
+  Node* adapt(std::shared_ptr<Graph> graph, Node* node) const override {
+    if (!HasAxisInput(node)) {
+      node->i_(kaxis, this->default_axis);
+      return EnsureAndReturnNode(node);
+    }
+
+    const ArrayRef<Value*>& inputs = node->inputs();
```

Comment:
Handle when the input value is unknown (fail)

Reply:
So, this is still TODO?